src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -146,446 +73,51 @@ object EthService {
-  case class GetUncleCountByBlockHashResponse(result: BigInt)
-
-  case class GetCoinbaseRequest()
-  case class GetCoinbaseResponse(address: Address)
-
-  case class GetBlockTransactionCountByNumberRequest(block: BlockParam)
-  case class GetBlockTransactionCountByNumberResponse(result: BigInt)
-
-  case class GetBalanceRequest(address: Address, block: BlockParam)
-  case class GetBalanceResponse(value: BigInt)
-
-  case class GetStorageAtRequest(address: Address, position: BigInt, block: BlockParam)
-  case class GetStorageAtResponse(value: ByteString)
-
-  case class GetTransactionCountRequest(address: Address, block: BlockParam)
-  case class GetTransactionCountResponse(value: BigInt)
-
   case class ResolvedBlock(block: Block, pendingState: Option[InMemoryWorldStateProxy])
 
-  case class NewFilterRequest(filter: Filter)
-  case class Filter(
-      fromBlock: Option[BlockParam],
-      toBlock: Option[BlockParam],
-      address: Option[Address],
-      topics: Seq[Seq[ByteString]]
-  )
-
-  case class NewBlockFilterRequest()
-  case class NewPendingTransactionFilterRequest()
-
-  case class NewFilterResponse(filterId: BigInt)
-
-  case class UninstallFilterRequest(filterId: BigInt)
-  case class UninstallFilterResponse(success: Boolean)
-
-  case class GetFilterChangesRequest(filterId: BigInt)
-  case class GetFilterChangesResponse(filterChanges: FilterChanges)
-
-  case class GetFilterLogsRequest(filterId: BigInt)
-  case class GetFilterLogsResponse(filterLogs: FilterLogs)
-
-  case class GetLogsRequest(filter: Filter)
-  case class GetLogsResponse(filterLogs: LogFilterLogs)
-
-  case class GetStorageRootRequest(address: Address, block: BlockParam)
-  case class GetStorageRootResponse(storageRoot: ByteString)
+  def resolveBlock(
+      blockchain: Blockchain,
+      ledger: Ledger,
+      blockParam: BlockParam
+  ): Either[JsonRpcError, ResolvedBlock] = {
+    def getBlock(number: BigInt): Either[JsonRpcError, Block] = {
+      blockchain
+        .getBlockByNumber(number)
+        .map(Right.apply)
+        .getOrElse(Left(JsonRpcError.InvalidParams(s"Block $number not found")))
+    }
 
-  case class EthPendingTransactionsRequest()
-  case class EthPendingTransactionsResponse(pendingTransactions: Seq[PendingTransaction])
+    blockParam match {
+      case BlockParam.WithNumber(blockNumber) => getBlock(blockNumber).map(ResolvedBlock(_, pendingState = None))
+      case BlockParam.Earliest => getBlock(0).map(ResolvedBlock(_, pendingState = None))
+      case BlockParam.Latest => getBlock(blockchain.getBestBlockNumber()).map(ResolvedBlock(_, pendingState = None))
+      case BlockParam.Pending =>
+        ledger.consensus.blockGenerator.getPendingBlockAndState
+          .map(pb => ResolvedBlock(pb.pendingBlock.block, pendingState = Some(pb.worldState)))
+          .map(Right.apply)
+          .getOrElse(resolveBlock(blockchain, ledger, BlockParam.Latest)) //Default behavior in other clients
+    }
+  }
 }
 
 class EthService(
```

Comment:
How about we move `call` and `ieleCall` to `EthTxService` (they are related to transactions right?) so that only info-related methods are left and we call this `EthInfoService` or `EthStatusService`?

Reply:
Considering that `call` and `ieleCall` don't actually alter anything (please correct me if I'm wrong) perhaps they could also be part of the `EthInfoService`; they provide info about smart contracts (namely the output.)